staging/src/k8s.io/apiserver/pkg/authentication/authenticator/interfaces.go

```diff
@@ -29,6 +30,71 @@ type Token interface {
 	AuthenticateToken(ctx context.Context, token string) (*Response, bool, error)
 }
 
+// WrapInInstrumentedToken wraps the input Token in the InstrumentedRequest.
+func WrapInInstrumentedToken(token Token, authenticatorName string) *InstrumentedToken {
+	return &InstrumentedToken{AuthenticatorName: authenticatorName, delegate: token}
+}
+
+// InstrumentedToken wraps Token and allows instrumentation to identify authenticators via the AuthenticatorName field.
+type InstrumentedToken struct {
```

Comment:
Does this type need to be exported?

Reply:
No (same for the other instrumented types) - unexported.